src/libmain/progress-bar.cc

```diff
@@ -480,12 +480,12 @@ class ProgressBar : public Logger
     }
 };
 
-Logger * makeProgressBar(bool printBuildLogs)
+std::unique_ptr<Logger> makeProgressBar(bool printBuildLogs)
 {
-    return new ProgressBar(
+    return std::unique_ptr<Logger>(new ProgressBar(
```

Comment:
std::make_unique please. The current proposal has a race condition where throwing as part of the ProgressBar constructor would still leak memory.